File: docs/demo/lobs/oci/fupdate.c
/* This file is installed in the following path when you install */
/* the database: $ORACLE_HOME/rdbms/demo/lobs/oci/fupdate.c */

/* Updating a BFILE by initializing a BFILE locator. */
#include <oratypes.h>
#include "lobdemo.h"
void BfileUpdate_proc(OCILobLocator *Bfile_loc, OCIEnv *envhp,
                      OCIError *errhp, OCISvcCtx *svchp, OCIStmt *stmthp)
{ 
  OCIBind *bndhp, *bndhp2; 

  text  *updstmt =  
    (text *) "UPDATE Print_media SET ad_graphic = :Lob_loc \
              WHERE product_id = 3107 AND ad_id = 13002"; 

  OraText *Dir = (OraText *)"MEDIA_DIR", 
          *Name = (OraText *)"keyboard_logo.jpg"; 

  printf ("----------- OCI BFILE Update Demo --------------\n"); 
  /* Prepare the SQL statement: */ 
  checkerr (errhp, OCIStmtPrepare(stmthp, errhp, updstmt,  (ub4)  
                                  strlen((char *) updstmt), 
                                  (ub4) OCI_NTV_SYNTAX, (ub4)OCI_DEFAULT)); 
 
  checkerr (errhp, OCILobFileSetName(envhp, errhp, &Bfile_loc, 
                                     Dir, (ub2)strlen((char *)Dir), 
                                     Name,(ub2)strlen((char *)Name))); 
 
  checkerr (errhp, OCIBindByPos(stmthp, &bndhp, errhp, (ub4) 1, 
                                (void *) &Bfile_loc, (sb4) 0,  SQLT_BFILE, 
                                (void *) 0, (ub2 *)0, (ub2 *)0, 
                                (ub4) 0, (ub4 *) 0, (ub4) OCI_DEFAULT));
 
 /* Execute the SQL statement: */ 
  checkerr (errhp, OCIStmtExecute(svchp, stmthp, errhp, (ub4) 1, (ub4) 0, 
                                  (CONST OCISnapshot*) 0, (OCISnapshot*) 0,   
                                  (ub4) OCI_DEFAULT)); 
  printf("Bfile column updated \n");
}

File: docs/demo/lobs/java/fupdate.java
/* This file is installed in the following path when you install */
/* the database: $ORACLE_HOME/rdbms/demo/lobs/java/fupdate.java */

/* Updating a BFILE column. 
 * Uses Oracle proprietary classes.
 * In JDK5 invoke with java -Djdbc.drivers=oracle.jdbc.OracleDriver fupdate
*/

import java.sql.Connection;
import java.sql.Statement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

import oracle.sql.BFILE;
import oracle.jdbc.OracleResultSet;
import oracle.jdbc.OraclePreparedStatement;

public class fupdate
{
  public static void main (String args [])
    throws Exception
  {
    Connection conn = LobDemoConnectionFactory.getConnection();
    Statement stmt = conn.createStatement ();
    ResultSet rset = stmt.executeQuery (
      "SELECT ad_graphic FROM Print_media" 
      + " WHERE product_id = 3106 AND ad_id = 13001");
    if (rset.next())
    {
      BFILE bfile = ((OracleResultSet)rset).getBFILE (1);

      PreparedStatement pstmt = conn.prepareStatement (
        "UPDATE Print_media SET ad_graphic = ?" 
        + " WHERE product_id = 3060 AND ad_id = 11001");
      ((OraclePreparedStatement)pstmt).setBFILE(1, bfile);
      pstmt.execute();
      System.out.println( "PRINT_MEDIA row updated with BFILE" );
      pstmt.close();
    }
    stmt.close();
    conn.commit();
    conn.close();
  }
}
